aiosmtpd/smtp.py

```diff
@@ -24,30 +25,44 @@
 from warnings import warn
 
 
-__version__ = '1.2.2'
-__ident__ = 'Python SMTP {}'.format(__version__)
-log = logging.getLogger('mail.log')
-
-
-DATA_SIZE_DEFAULT = 33554432
-EMPTYBYTES = b''
-NEWLINE = '\n'
-
+# region #### Custom Data Types #######################################################
 
 class _Missing:
     pass
 
 
-MISSING = _Missing()
+class _AuthMechAttr(NamedTuple):
+    method: "AuthMechanismType"
+    is_builtin: bool
+
+
+class _DataState(enum.Enum):
+    NOMINAL = enum.auto()
+    TOO_LONG = enum.auto()
+    TOO_MUCH = enum.auto()
 
 
 AuthMechanismType = Callable[["SMTP", List[str]], Awaitable[Any]]
 _TriStateType = Union[None, _Missing, bytes]
 
 
-class _AuthMechAttr(NamedTuple):
-    method: AuthMechanismType
-    is_builtin: bool
+# endregion
+
+
+# region #### Constant & Constant-likes ###############################################
+
+__version__ = '1.2.3a7'
+__ident__ = 'Python SMTP {}'.format(__version__)
+log = logging.getLogger('mail.log')
+
+
+DATA_SIZE_DEFAULT = 33_554_432  # Where does this number come from, I wonder...
```

Comment:
Maybe it would be better to write `2**25` (because that's what that number is)?

PS. Thanks for picking this up!

Reply:
It was?? \*grabs a calculator\* ... well, I'll be ... 😄

That's a good idea to write it `2**25` ... but where does that come from? I mean, is it a buffer limitation somewhere?

Reply:
Not a very important question, though. So I'll resolve this.